backend/app/services/file_service.py

```
@@ -513,4 +514,178 @@ async def process_uploaded_file(file: UploadFile) -> str:
+        images_dir = os.path.join(settings.upload_dir, "material_images", material_id)
+        os.makedirs(images_dir, exist_ok=True)
+        filename = f"image_{index:03d}.{ext}"
+        file_path = os.path.join(images_dir, filename)
+        with open(file_path, "wb") as f:
+            f.write(image_data)
+        # 返回相对于 upload_dir 的路径，方便后续 API 暴露
+        return os.path.join("material_images", material_id, filename)
+
+    @staticmethod
+    def _extract_pdf_with_images(file_path: str, material_id: str) -> Dict[str, Any]:
+        """从 PDF 提取文本和图片（分离模式），返回 {text, images}"""
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)